metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/extractor/ReferenceSequenceExtractor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.metadata.importer.extractor;
+import org.springframework.core.convert.converter.Converter;
+import uk.ac.ebi.ampt2d.metadata.importer.SraXmlRetrieverByAccession;
+import uk.ac.ebi.ampt2d.metadata.importer.xml.SraXmlParser;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.ReferenceSequence;
+import uk.ac.ebi.ena.sra.xml.AnalysisType;
+import uk.ac.ebi.ena.sra.xml.AssemblyType;
+import uk.ac.ebi.ena.sra.xml.ReferenceSequenceType;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ReferenceSequenceExtractor {
+
+    private static final Logger REFERENCE_SEQUENCE_EXTRACT_SERVICE_LOGGER = Logger.getLogger(
+            ReferenceSequenceExtractor.class.getName());
+
+    private SraXmlRetrieverByAccession sraRetrieverByAccession;
+    private SraXmlParser<AnalysisType> sraAnalysisXmlParser;
+    private SraXmlParser<AssemblyType> sraAssemblyXmlParser;
+    private Converter<AssemblyType, ReferenceSequence> referenceSequenceConverter;
+
+    public ReferenceSequenceExtractor(
+            SraXmlRetrieverByAccession sraRetrieverByAccession,
+            SraXmlParser<AnalysisType> sraAnalysisXmlParser,
+            SraXmlParser<AssemblyType> sraAssemblyXmlParser,
+            Converter<AssemblyType, ReferenceSequence> referenceSequenceConverter) {
+        this.sraRetrieverByAccession = sraRetrieverByAccession;
+        this.sraAnalysisXmlParser = sraAnalysisXmlParser;
+        this.sraAssemblyXmlParser = sraAssemblyXmlParser;
+        this.referenceSequenceConverter = referenceSequenceConverter;
+    }
+
+    public ReferenceSequence extractReferenceSequence(String analysisAccession) {
```

Comment:
can you please move this code to SraObjectsImporterThroughApi

Reply:
Done in https://github.com/EBIvariation/metadata-ws/pull/50/commits/d504b323f7313c1800a3ba0d5872018c12644af0 and https://github.com/EBIvariation/metadata-ws/pull/50/commits/c8d1eb058b4838fbe58325440d5224047036a339